algorithms/kernel/classifier/classifier_train.cpp

```diff
@@ -100,11 +102,30 @@ services::Status Input::check(const daal::algorithms::Parameter *parameter, int
 
 services::Status Input::checkImpl(const daal::algorithms::Parameter *parameter) const
 {
-    services::Status s;
+    services::Status s;  // Error status
+    int nClasses{0};     // Number of classes
+    bool flag{false};    // Flag indicates error in table of labels
+
     if (parameter != NULL)
     {
-        const Parameter *algParameter = static_cast<const Parameter *>(parameter);
-        DAAL_CHECK_EX(algParameter->nClasses > 1, services::ErrorIncorrectParameter, services::ParameterName, nClassesStr());
+        const daal::algorithms::classifier::interface1::Parameter *algParameter1 =
+            dynamic_cast<const daal::algorithms::classifier::interface1::Parameter *>(parameter);
+        const daal::algorithms::classifier::interface2::Parameter *algParameter2 =
+            dynamic_cast<const daal::algorithms::classifier::interface2::Parameter *>(parameter);
+        if (algParameter1 != NULL)
+        {
+            DAAL_CHECK_EX(algParameter1->nClasses > 1, services::ErrorIncorrectParameter, services::ParameterName, nClassesStr());
+            nClasses = 0; // Check correctness of label doesn't perform for deprecated interface 1
+        }
+        else if ((algParameter2 != NULL) && (INT_MAX > algParameter2->nClasses))
```

Comment:
ErrorNullParameterNotSupported will be return in case when INT_MAX < algParameter2->nClasses

Reply:
fixed